### DISCUSSION TITLE
Complete Vue3 monorepo

### DISCUSSION THREAD

**Discussion by JFGHT** [vercel/turborepo#2452] [Help] [answered]
I have tried to follow the similar Turborepo approach for Next and [this one](https://github.com/arneesh/turborepo-vue-starter) for Vue in order to combine it to have two Vue3 apps created via `create-vue` but with almost all options [Typescript, No JSX, Vitest, Router, Pinia, Cypress, ESLint, Prettier], but the setup has been more than hard.

I started configuring a shared Tsconfig which resulted in a nightmare. Files the `create-vue` creates:
`tsconfig.json`
```json
{
  "files": [],
  "references": [
    {
      "path": "tsconfig/vueapp/tsconfig.config.json"
    },
    {
      "path": "tsconfig/vueapp/tsconfig.app.json"
    },
    {
      "path": "tsconfig/vueapp/tsconfig.vitest.json"
    }
  ]
}
````
`tsconfig.app.json`
```json
{
  "extends": "@vue/tsconfig/tsconfig.web.json",
  "include": ["env.d.ts", "src/**/*", "src/**/*.vue"],
  "exclude": ["src/**/__tests__/*"],
  "compilerOptions": {
    "composite": true,
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```
`tsconfig.config.json`
```json
{
  "extends": "@vue/tsconfig/tsconfig.node.json",
  "include": ["vite.config.*", "vitest.config.*", "cypress.config.*", "playwright.config.*"],
  "compilerOptions": {
    "composite": true,
    "types": ["node"]
  }
}
```
`tsconfig.vitest.json`
```json
{
  "extends": "./tsconfig.app.json",
  "exclude": [],
  "compilerOptions": {
    "composite": true,
    "lib": [],
    "types": ["node", "jsdom"]
  }
}

```

What I've done so far:

1.- Created the two Vue3 apps inside apps.
2.- Created a file called `vueapps` inside `packages/tsconfig` moving these three json files from above except for the main tsconfig.
3.- Installed `@vue/tsconfig` inside `vueapps`.
4.- Included into `packages/package.json` in the files property:
```json
"files": [
    "base.json",
    "vueapp/tsconfig.app.json",
    "vueapp/tsconfig.config.json",
    "vueapp/tsconfig.vitest.json"
  ],
```
5.- Added into every vue app in `devDependencies` the line: `"tsconfig": "workspace:*",`.

And here is when the problems begin.

- Leaving the `tsconfig.json` as it is in the app does nothing.
- Modifying `tsconfig.json` just leaving it with:
```json
"extends": "tsconfig/vueapp/tsconfig.app.json"
```
Throws:
```
No inputs were found in config file '/home/jfght/Documentos/apps/projectname/apps/bloqadmin/tsconfig.json'. Specified 'include' paths were '["../../packages/tsconfig/vueapp/env.d.ts","../../packages/tsconfig/vueapp/src/**/*","../../packages/tsconfig/vueapp/src/**/*.vue"]' and 'exclude' paths were '["../../packages/tsconfig/vueapp/src/**/__tests__/*"]'.
```
Something that does not happen when adding `"include": ["env.d.ts", "src/**/*", "src/**/*.vue"],` directly without the `extends`. As you can see, it's adding the path `../../packages/tsconfig/vueapp`. same for the excludes.

I've been trying to have the complete thing setup but I couldn't. Any idea?

**Comment by JFGHT** [comment]
Suggestions?

**Comment by anthonyshew** [comment, marked as answer]
We have added a Vue3/Nuxt example in the examples directory: https://github.com/vercel/turbo/tree/main/examples/with-vue-nuxt
